DXMainClient/Program.cs

```diff
@@ -116,25 +129,71 @@ static void Main(string[] args)
                     mutex.ReleaseMutex();
             }
         }
-#if !DEBUG
-
+ 
+#if !NETFRAMEWORK
         private static Assembly DefaultAssemblyLoadContextOnResolving(AssemblyLoadContext assemblyLoadContext, AssemblyName assemblyName)
         {
             if (assemblyName.Name.EndsWith(".resources", StringComparison.OrdinalIgnoreCase))
                 return null;
 
+            // the specific dll should be in priority than the common one
+
+            var specificFileInfo = new FileInfo(Path.Combine(SPECIFIC_LIBRARY_PATH, FormattableString.Invariant($"{assemblyName.Name}.dll")));
+
+            if (specificFileInfo.Exists)
+                return assemblyLoadContext.LoadFromAssemblyPath(specificFileInfo.FullName);
+
             var commonFileInfo = new FileInfo(Path.Combine(COMMON_LIBRARY_PATH, FormattableString.Invariant($"{assemblyName.Name}.dll")));
 
             if (commonFileInfo.Exists)
                 return assemblyLoadContext.LoadFromAssemblyPath(commonFileInfo.FullName);
 
-            var specificFileInfo = new FileInfo(Path.Combine(SPECIFIC_LIBRARY_PATH, FormattableString.Invariant($"{assemblyName.Name}.dll")));
+            return null;
+        }
+#else
+        private static Assembly CurrentDomain_AssemblyResolve(object sender, ResolveEventArgs args)
+        {
+            string unresolvedAssemblyName = args.Name.Split(',').First();
+
+            if (unresolvedAssemblyName.EndsWith(".resources", StringComparison.OrdinalIgnoreCase))
+                return null;
+
+            // the specific dll should be in priority than the common one
+
+            var specificFileInfo = new FileInfo(FormattableString.Invariant($"{Path.Combine(SPECIFIC_LIBRARY_PATH, unresolvedAssemblyName)}.dll"));
 
             if (specificFileInfo.Exists)
-                return assemblyLoadContext.LoadFromAssemblyPath(specificFileInfo.FullName);
+                return Assembly.Load(AssemblyName.GetAssemblyName(specificFileInfo.FullName));
+
+            var commonFileInfo = new FileInfo(FormattableString.Invariant($"{Path.Combine(COMMON_LIBRARY_PATH, unresolvedAssemblyName)}.dll"));
+
+            if (commonFileInfo.Exists)
+                return Assembly.Load(AssemblyName.GetAssemblyName(commonFileInfo.FullName));
 
             return null;
         }
 #endif
+
+        // SearchResourcesDir is copied from ClientCore
+        private static string SearchResourcesDir(string startupPath)
```

Comment:
I think should add the explanation for this and a cref to the original method to a docstring so that if someone updates one of the methods the second would need to be updated too, and a comment in the second place backreferencing that one.